src/korma/core.clj

```diff
@@ -533,11 +533,14 @@
 (defn exec
   "Execute a query map and return the results."
   [query]
-  (if (or (seq (:pre-side-effects query)) (seq (:post-side-effects query))
+  (if (or (seq (get-in query [:ent :pre-side-effects]))
+          (seq (get-in query [:ent :post-side-effects]))
           (-> query :ent :prepares seq))
-    (db/with-db (or db/*current-db* (get query :db))
-      (db/transaction
-        (exec* query)))
+    (do (when-not db/*current-conn*
```

Comment:
Would this work instead? It avoids changing the global default connection, and I think since it reuses the existing connection it would preserve the desired behavior:

```clj
(binding [*current-conn* (or *current-conn* (get-connection (:db query))]
  (db/transaction
    (exec* query))
```

The global default connection is a sharp edge in Korma (though the dynamic `*current-conn*` and `*current-db*` aren't much better) because we use Korma with multiple databases (e.g., pinpoint and analytics--this is why we have the convention of manually setting the default connection to the desired database before a transaction to avoid surprises).

Reply:
You're right, I shouldn't touch the default connection. I actually needed to mirror their code:

https://github.com/liftoffio/Korma/blob/e2a2c48135856ee69b382d64add0886234484b38/src/korma/db.clj#L301-L305

I updated the code.